lib/errors.js

```diff
@@ -9,18 +9,48 @@ const codes = {
   INVALID_DELETION_OPERATION: 1004,
   INVALID_CREATION_OPERATION: 1005,
   INSUFFICIENT_PERMISSIONS: 1006,
+  ACTION_EXECUTION_ERROR: 1007,
 };
 
 const defaultMessages = {
   [codes.NOT_IMPLEMENTED]: 'Not implemented',
 };
 
+class ActionError extends Error {
+  constructor(name, ctx) {
```

Comment:
```suggestion
  constructor(message, ctx) {
```

Reply:
That's not a message, but an error name.